### PR TITLE
rustc: bump to 1.29.1

### PR DESCRIPTION
**Pull request by alarcher**
@Mno-hime If you have time to review.

### CODE REVIEW

File: components/developer/rust/rustc.p5m [hunk 1]
```diff
@@ -22,8 +22,9 @@ set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
 
 license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
 
-<transform file path=usr/lib/rustlib/x86_64-sun-solaris/lib/.*.so.* -> default pkg.linted.userland.action001.2 true >
-<transform file path=usr/lib/rustlib/x86_64-sun-solaris/codegen-backends/.*.so.* -> default pkg.linted.userland.action001.2 true >
+<transform file path=usr/lib/rustlib/$(RUST_ARCH)/lib/.*.so.* -> default pkg.linted.userland.action001.2 true >
+<transform file path=usr/lib/rustlib/$(RUST_ARCH)/lib/.*.so.* -> default pkg.linted.userland.action001.3 true >
+<transform file path=usr/lib/rustlib/$(RUST_ARCH)/codegen-backends/.*.so.* -> default pkg.linted.userland.action001.2 true >
```
Comment (Toasterson) [inline components/developer/rust/rustc.p5m:27]: Please add a comment here what these transforms are about. Helps when later somebody needs to figure out what the history of these lines where. So one can evaluate if the lines are needed still or not.

File: components/developer/rust/Makefile [hunk 2]
```diff
@@ -139,12 +139,33 @@ COMPONENT_INSTALL_ENV += DESTDIR=$(PROTO_DIR)
...
+RUSTC_BIN_POST+= rls
+RUSTC_BIN_POST+= rustc
+RUSTC_BIN_POST+= rustdoc
+
 COMPONENT_POST_INSTALL_ACTION= \
-	( $(RM) $(PROTOUSRLIBDIR)/*.so; \
-	  $(MKDIR) $(PROTOUSRLIBDIR.$(BITS)); \
-	  $(RM) $(PROTOUSRLIBDIR.$(BITS))/*.so; \
-	  cd $(PROTOUSRLIBDIR.$(BITS)); \
-	  $(LN) -s ../rustlib/$(RUST_ARCH)/lib/*.so .; )
```
Comment (Toasterson) [inline components/developer/rust/Makefile:147]: Please add a comment about the History/Reason of the Post Install Changes. This helps other evaluate if they are still needed or not, in later versions of the rust Package

**Review by Toasterson — COMMENTED** [review]
Apart from the Two point where some comments would help later packaging work LGTM

File: components/developer/rust/rustc.p5m (hunk 1 above)
Comment (alarcher) [reply components/developer/rust/rustc.p5m:27]: Fixed.

File: components/developer/rust/Makefile (hunk 2 above)
Comment (alarcher) [reply components/developer/rust/Makefile:147]: Fixed.

**Review by pyhalov — APPROVED** [review]

**Review by Toasterson — APPROVED** [review]

**Pull request merged**